mempool/mempool.go

```diff
@@ -675,8 +720,8 @@ func (cache *mapTxCache) Push(tx types.Tx) bool {
 			cache.list.Remove(popped)
 		}
 	}
-	cache.list.PushBack(txHash)
-	cache.map_[txHash] = cache.list.Back()
+	e := cache.list.PushFront(txHash)
```

Comment:
Ok I think something is wrong above. If the tx is already in the cache, we move it to the front, and if its new, it should be added to the front (thanks for the fix here). But if the cache is full, it seems we pop the element from the front - shouldn't we take that from the back?

Reply:
Oh you're right, great catch! I'll also add a cache test for the LRU property.

Reply:
Noticed that too https://github.com/tendermint/tendermint/pull/3036 today. Issue https://github.com/tendermint/tendermint/issues/3035